homeassistant/components/omnilogic/switch.py

```diff
@@ -0,0 +1,256 @@
+                    icon=entity_setting["icon"],
+                )
+
+                entities.append(entity)
+
+    async_add_entities(entities)
+
+    # register service
+    platform = entity_platform.current_platform.get()
+
+    platform.async_register_entity_service(
+        SERVICE_SET_SPEED,
+        {vol.Required("speed"): cv.positive_int},
+        "async_set_speed",
+    )
+
+
+class OmniLogicSwitch(OmniLogicEntity, SwitchEntity):
+    """Define an Omnilogic Switch entity."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        icon: str,
+        item_id: tuple,
+        state_key: str,
+    ):
+        """Initialize Entities."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            item_id=item_id,
+            icon=icon,
+        )
+
+        self._state_key = state_key
+        self._state = None
+        self._last_action = 0
+        self._state_delay = 30
+
+    @property
+    def is_on(self):
+        """Return the on/off state of the switch."""
+        if self._last_action < (time.time() - self._state_delay):
+            self._state = int(self.coordinator.data[self._item_id][self._state_key])
+
+            if self._state == 7:
+                self._state = 0
+
+        return self._state
+
+
+class OmniLogicRelayControl(OmniLogicSwitch):
+    """Define the OmniLogic Relay entity."""
+
+    async def async_turn_on(self, **kwargs):
+        """Turn on the relay."""
+        self._state = True
+        self._last_action = time.time()
+        self.async_schedule_update_ha_state()
+
+        await self.coordinator.api.set_relay_valve(
+            int(self._item_id[1]),
+            int(self._item_id[3]),
+            int(self._item_id[-1]),
+            1,
+        )
+
+    async def async_turn_off(self, **kwargs):
+        """Turn off the relay."""
+        self._state = False
+        self._last_action = time.time()
+        self.async_schedule_update_ha_state()
+
+        await self.coordinator.api.set_relay_valve(
+            int(self._item_id[1]),
+            int(self._item_id[3]),
+            int(self._item_id[-1]),
+            0,
+        )
+
+
+class OmniLogicPumpControl(OmniLogicSwitch):
+    """Define the OmniLogic Pump Switch Entity."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        icon: str,
+        item_id: tuple,
+        state_key: str,
+    ):
+        """Initialize entities."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            icon=icon,
+            item_id=item_id,
+            state_key=state_key,
+        )
+
+        self._max_speed = int(coordinator.data[item_id]["Max-Pump-Speed"])
+        self._min_speed = int(coordinator.data[item_id]["Min-Pump-Speed"])
```

Comment:
Can these change?

Reply:
No - the min and max pump speed are specific to a particular pump and do not change.